File: src/Rook.java
import java.util.List;

public class Rook extends Piece{
    public Rook(Color color) {
        super(color, 5,"♖","♜");
    }

    //Handle castling
    @Override
    public List<List<Event>> getValidMoves(Board board, Position position) {
        return board.getAllStraightMoves(position);
    }
}

File: src/pieces/Rook.kt
package pieces

import Board
import Color
import Piece
import Position
import Tempo

class Rook(color: Color) : Piece(color, 5, "♜") {
    //Handle castling
    override fun getValidMoves(board: Board, position: Position, checkValidation: Boolean): List<Tempo> {
        return board.getAllStraightMoves(position, checkValidation)
    }

    override fun couldBeCheck(board: Board, from: Position, to: Position): Boolean {
        return from.row == to.row || from.col == to.col
    }
}